api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java

```diff
@@ -299,4 +298,85 @@ public BedLocationMapping getBedLocationMappingByBed(Bed bed) {
         query.setParameter("bed", bed);
         return (BedLocationMapping) query.uniqueResult();
     }
+
+    private Criteria createGetBedsCriteria(Location location, BedType bedType, BedStatus bedStatus, Integer limit, Integer offset) {
+        Session session = sessionFactory.getCurrentSession();
+        Criteria criteria;
+        if (location != null) {
+            criteria = session.createCriteria(BedLocationMapping.class, "blm");
+            criteria.createAlias("blm.bed", "bed");
+            criteria.createAlias("blm.location", "location");
+            criteria.add(Restrictions.eq("location", location));
```

Comment:
@sanishmaharjan can you not make the restriction on `location` conditional (depending on whether `location` is null or not) and always use the criteria based on `BedLocationMapping.class`?

Reply:
Whether a location is null or not it will not retrieve beds that not associated with BedLocationMapping.
Here bed with bed_id="17" is not associated with BedLocationMapping so it will not retrieve if we always use the criteria base on BedLocationMapping.class.

https://github.com/openmrs/openmrs-module-bedmanagement/blob/036df94c620f8ab6d7b7f571c55a0a5a53713973/api/src/test/resources/bedManagementDAOComponentTestDataset.xml#L54-L68

Lets if we try to retrieve beds with status "AVAILABLE" then it will not retrieve bed with bed_id="17" even though its status is "AVAILABLE".

Reply:
Thanks, that's very clear.